codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java

```diff
@@ -176,44 +182,54 @@ private static MqttFixedHeader decodeFixedHeader(ByteBuf buffer) {
      * @param mqttFixedHeader MqttFixedHeader of the same message
      * @return the variable header
      */
-    private static Result<?> decodeVariableHeader(ByteBuf buffer, MqttFixedHeader mqttFixedHeader) {
+    private Result<?> decodeVariableHeader(ChannelHandlerContext ctx, ByteBuf buffer, MqttFixedHeader mqttFixedHeader) {
         switch (mqttFixedHeader.messageType()) {
             case CONNECT:
-                return decodeConnectionVariableHeader(buffer);
+                return decodeConnectionVariableHeader(ctx, buffer);
 
             case CONNACK:
-                return decodeConnAckVariableHeader(buffer);
+                return decodeConnAckVariableHeader(ctx, buffer);
 
-            case SUBSCRIBE:
             case UNSUBSCRIBE:
+            case SUBSCRIBE:
             case SUBACK:
             case UNSUBACK:
+                return decodeMessageIdAndPropertiesVariableHeader(ctx, buffer);
+
             case PUBACK:
             case PUBREC:
             case PUBCOMP:
             case PUBREL:
-                return decodeMessageIdVariableHeader(buffer);
+                return decodePubReplyMessage(buffer);
 
             case PUBLISH:
-                return decodePublishVariableHeader(buffer, mqttFixedHeader);
+                return decodePublishVariableHeader(ctx, buffer, mqttFixedHeader);
+
+            case DISCONNECT:
+            case AUTH:
+                return decodeReasonCodeAndPropertiesVariableHeader(buffer);
```

Comment:
Given nothing was parsed in the past for old mqtt protocol is this change compatible?

Reply:
`decodeReasonCodeAndPropertiesVariableHeader` checks for `bytesRemainingInVariablePart`. If there's no bytes remaining it will substitute default values in `MqttReasonCodeAndPropertiesVariableHeader` (reason code `0` and no properties).

Reply:
@johnou I think this should work... any concerns ?

Reply:
@johnou please resolve if you agree and if not leave another comment

Reply:
Cannot resolve, change was force pushed.

Reply:
But yes I agree, should be okay.